### ISSUE TITLE
VRML/X3D-parser does *not* report errors anymore

### ISSUE THREAD

**Issue by elmkni** [castle-engine/view3dscene#53]
Hi Michalis,

it's me, Elmar (again).

I recently stumbled over a **parser issue** in the newest [View3DScene](https://jenkins.castle-engine.io/public/builds/view3dscene/view3dscene-4.3.0-win64-x86_64.zip) from 25-Apr-2023 23:12 (and probably earlier versions):

**It seems that the VRML/X3D parser does not report errors anymore when reading input files.**

The file is loaded until the error - everything after it is discarded ...

Here is a simple test-case:
```
#X3D V4.0 utf8

Shape {
 appearance Appearance {
  material Material {
   diffuseColor 1 0 
  }
 }
 geometry Box { size 1 2 3 }
}
```
The cube doesn't show up, no error is reported.

The errors in this simple test-case may be obvious (the missing PROFILE statement [which is actually ignored, even though it is compulsory] and the missing third value for SFColor [for diffuseColor] ), but in a more complex scenario it is hard to find an error without _a helping hand_ from the parser!

Was the old behaviour deliberately dropped for some reason, or is it a bug?

I would rather prefer the old (nagging) behaviour, because it forces me to write better code!

Please help!

With regards,

Elmar

**Comment by michaliskambi** [comment]
We absolutely want to display the parsing errors, and we do -- testing on your testcase, latest view3dscene shows "2 warnings" and clicking on the button shows them correctly. See the screenshot.

![Zrzut ekranu z 2023-04-26 17-34-35](https://user-images.githubusercontent.com/1076978/234627519-263d8f27-3a9b-42c7-9f8b-2c369f3b1394.png)

Are you sure you don't see them? Note that closing this dialog makes the "2 warnings" button disappear. It will only appear again if *new* warnings will happen.  This way new warnings will "pop up" if something new occurred. But all warnings are always available if you use "File->View Warnings About Current Scene". It was like that since many years (that the "XXX warnings" button disappears) though maybe we could change it.

In any case:

- We haven't changed here anything recently, we show parsing errors as warnings. And of course we want to keep showing them, just as you say -- this is valuable for user to fix the faulty scene.

- Please confirm you can see them too :) If not, please record a movie how you open the model, maybe you use view3dscene in some way I don't expect :)

**Comment by elmkni** [comment]
Hi Michalis,

you wrote:

> Please confirm you can see them too :) If not, please record a movie how you open the model, maybe you use view3dscene in some way I don't expect :)

### Actually, no warning button pops up (as in the older versions) when I load the faulty scene! 
### I have to explicitly use the menu `View Warnings About Current Scene` to see the warnings!
### The old View3DScene from October 2022 does show the warning button though! --- Strange ...

---

It took me some time to record a video of the issue, because [`XBox Game Bar`](https://apps.microsoft.com/store/detail/xbox-game-bar/9NZKPSTSNW4P) (that I used to record the other [videos](https://github.com/castle-engine/view3dscene/files/11290907/v4_test_scale_rotate.zip)) is unable to record drop-down menus!

I had to test 5 programs before I found [ActivePresenter 9](https://atomisystems.com) that did the trick ...

---

### Here is the zipped video + testcase file: [view3dscene_issue_53.zip](https://github.com/castle-engine/view3dscene/files/11337360/view3dscene_issue_53.zip)

---

It is really strange, that I cannot see the warnings, but you can! 

I have not the slightest idea what causes the difference, 

I hope you, in your infinite wisdom, can find the reason for the problem ...

With regards,

Elmar

 ---

I use the 64-bit windows version of View3DScene on Windows 11:
```
Intel(R) Pentium(R) Silver N5030 CPU @ 1.10GHz   1.10 GHz
4,00 GB (3,81 GB usable), 64-Bit-OS, x64-based Processor
OS: Windows 11 Pro Education, 22H2, build 22621.1635
```
### P.S.: Here are the system informations from the View3DScene Help menu:

OpenGL information (detected by view3dscene):

----------------------------------------
Basic information:

  Context requested: automatic

  Version string: 4.6.0 - Build 30.0.101.1960
  Version parsed: major: 4, minor: 6, release exists: True, release: 0, vendor-specific information: "- Build 30.0.101.1960"
  Version in suffix, specific to vendor: major: 30, minor: 0, release: 101

  Vendor string: Intel
  Vendor type: Intel
  Special vendors:
    Fglrx: False
    Mesa: False

  Renderer: Intel(R) UHD Graphics 605

----------------------------------------
Confirmed versions:
(checks both version string and actual functions availability in the library, to secure from buggy implementations)

  1.2: True
  1.3: True
  1.4: True
  1.5: True
  2.0: True
  2.1: True
  3.0: True
  3.1: True
  3.2: True
  3.3: True
  4.0: True

----------------------------------------
Features:

  Shaders (GLSL): True
  Multi-texturing: True
  Framebuffer Object: True
  Multi-sampling for FBO buffers and textures: True
  Vertex Buffer Object: True
  Vertex Array Object: True
  GenerateMipmap available (and reliable): True
  Cube map textures: True
  Compressed textures supported: [DXT1_RGB, DXT1_RGBA, DXT3, DXT5, ASTC_4x4_RGBA, ASTC_5x4_RGBA, ASTC_5x5_RGBA, ASTC_6x5_RGBA, ASTC_6x6_RGBA, ASTC_8x5_RGBA, ASTC_8x6_RGBA, ASTC_8x8_RGBA, ASTC_10x5_RGBA, ASTC_10x6_RGBA, ASTC_10x8_RGBA, ASTC_10x10_RGBA, ASTC_12x10_RGBA, ASTC_12x12_RGBA, ASTC_4x4_SRGB8_ALPHA8, ASTC_5x4_SRGB8_ALPHA8, ASTC_5x5_SRGB8_ALPHA8, ASTC_6x5_SRGB8_ALPHA8, ASTC_6x6_SRGB8_ALPHA8, ASTC_8x5_SRGB8_ALPHA8, ASTC_8x6_SRGB8_ALPHA8, ASTC_8x8_SRGB8_ALPHA8, ASTC_10x5_SRGB8_ALPHA8, ASTC_10x6_SRGB8_ALPHA8, ASTC_10x8_SRGB8_ALPHA8, ASTC_10x10_SRGB8_ALPHA8, ASTC_12x10_SRGB8_ALPHA8, ASTC_12x12_SRGB8_ALPHA8]
  3D textures: True
  Textures non-power-of-2: True
  Blend constant parameter: True
  Float textures: True
  Depth textures: True
  Shadow samplers (necessary for shadow maps): True
  Packed depth + stencil: True
  Texture Modern Formats + Swizzle: True
  Shadow Volumes Possible (stencil buffer, two-sided stencil operations): True
  Occlusion Query: True
  Texture Max Level: True

  Necessary to use deprecated fixed-function pipeline: False

  All extensions: 244 extensions: (0) GL_3DFX_texture_compression_FXT1 (1) GL_AMD_depth_clamp_separate (2) GL_AMD_vertex_shader_layer (3) GL_AMD_vertex_shader_viewport_index (4) GL_ARB_ES2_compatibility (5) GL_ARB_ES3_1_compatibility (6) GL_ARB_ES3_compatibility (7) GL_ARB_arrays_of_arrays (8) GL_ARB_base_instance (9) GL_ARB_bindless_texture (10) GL_ARB_blend_func_extended (11) GL_ARB_buffer_storage (12) GL_ARB_cl_event (13) GL_ARB_clear_buffer_object (14) GL_ARB_clear_texture (15) GL_ARB_clip_control (16) GL_ARB_color_buffer_float (17) GL_ARB_compatibility (18) GL_ARB_compressed_texture_pixel_storage (19) GL_ARB_compute_shader (20) GL_ARB_conditional_render_inverted (21) GL_ARB_conservative_depth (22) GL_ARB_copy_buffer (23) GL_ARB_copy_image (24) GL_ARB_cull_distance (25) GL_ARB_debug_output (26) GL_ARB_depth_buffer_float (27) GL_ARB_depth_clamp (28) GL_ARB_depth_texture (29) GL_ARB_derivative_control (30) GL_ARB_direct_state_access (31) GL_ARB_draw_buffers (32) GL_ARB_draw_buffers_blend (33) GL_ARB_draw_elements_base_vertex (34) GL_ARB_draw_indirect (35) GL_ARB_draw_instanced (36) GL_ARB_enhanced_layouts (37) GL_ARB_explicit_attrib_location (38) GL_ARB_explicit_uniform_location (39) GL_ARB_fragment_coord_conventions (40) GL_ARB_fragment_layer_viewport (41) GL_ARB_fragment_program (42) GL_ARB_fragment_program_shadow (43) GL_ARB_fragment_shader (44) GL_ARB_fragment_shader_interlock (45) GL_ARB_framebuffer_no_attachments (46) GL_ARB_framebuffer_object (47) GL_ARB_framebuffer_sRGB (48) GL_ARB_geometry_shader4 (49) GL_ARB_get_program_binary (50) GL_ARB_get_texture_sub_image (51) GL_ARB_gl_spirv (52) GL_ARB_gpu_shader5 (53) GL_ARB_gpu_shader_fp64 (54) GL_ARB_half_float_pixel (55) GL_ARB_half_float_vertex (56) GL_ARB_indirect_parameters (57) GL_ARB_instanced_arrays (58) GL_ARB_internalformat_query (59) GL_ARB_internalformat_query2 (60) GL_ARB_invalidate_subdata (61) GL_ARB_map_buffer_alignment (62) GL_ARB_map_buffer_range (63) GL_ARB_multi_bind (64) GL_ARB_multi_draw_indirect (65) GL_ARB_multisample (66) GL_ARB_multitexture (67) GL_ARB_occlusion_query (68) GL_ARB_occlusion_query2 (69) GL_ARB_pipeline_statistics_query (70) GL_ARB_pixel_buffer_object (71) GL_ARB_point_parameters (72) GL_ARB_point_sprite (73) GL_ARB_polygon_offset_clamp (74) GL_ARB_post_depth_coverage (75) GL_ARB_program_interface_query (76) GL_ARB_provoking_vertex (77) GL_ARB_query_buffer_object (78) GL_ARB_robust_buffer_access_behavior (79) GL_ARB_robustness (80) GL_ARB_robustness_isolation (81) GL_ARB_sample_shading (82) GL_ARB_sampler_objects (83) GL_ARB_seamless_cube_map (84) GL_ARB_seamless_cubemap_per_texture (85) GL_ARB_separate_shader_objects (86) GL_ARB_shader_atomic_counter_ops (87) GL_ARB_shader_atomic_counters (88) GL_ARB_shader_bit_encoding (89) GL_ARB_shader_draw_parameters (90) GL_ARB_shader_group_vote (91) GL_ARB_shader_image_load_store (92) GL_ARB_shader_image_size (93) GL_ARB_shader_objects (94) GL_ARB_shader_precision (95) GL_ARB_shader_stencil_export (96) GL_ARB_shader_storage_buffer_object (97) GL_ARB_shader_subroutine (98) GL_ARB_shader_texture_image_samples (99) GL_ARB_shading_language_100 (100) GL_ARB_shading_language_420pack (101) GL_ARB_shading_language_packing (102) GL_ARB_shadow (103) GL_ARB_spirv_extensions (104) GL_ARB_stencil_texturing (105) GL_ARB_sync (106) GL_ARB_tessellation_shader (107) GL_ARB_texture_barrier (108) GL_ARB_texture_border_clamp (109) GL_ARB_texture_buffer_object (110) GL_ARB_texture_buffer_object_rgb32 (111) GL_ARB_texture_buffer_range (112) GL_ARB_texture_compression (113) GL_ARB_texture_compression_bptc (114) GL_ARB_texture_compression_rgtc (115) GL_ARB_texture_cube_map (116) GL_ARB_texture_cube_map_array (117) GL_ARB_texture_env_add (118) GL_ARB_texture_env_combine (119) GL_ARB_texture_env_crossbar (120) GL_ARB_texture_env_dot3 (121) GL_ARB_texture_filter_anisotropic (122) GL_ARB_texture_float (123) GL_ARB_texture_gather (124) GL_ARB_texture_mirror_clamp_to_edge (125) GL_ARB_texture_mirrored_repeat (126) GL_ARB_texture_multisample (127) GL_ARB_texture_non_power_of_two (128) GL_ARB_texture_query_levels (129) GL_ARB_texture_query_lod (130) GL_ARB_texture_rectangle (131) GL_ARB_texture_rg (132) GL_ARB_texture_rgb10_a2ui (133) GL_ARB_texture_stencil8 (134) GL_ARB_texture_storage (135) GL_ARB_texture_storage_multisample (136) GL_ARB_texture_swizzle (137) GL_ARB_texture_view (138) GL_ARB_timer_query (139) GL_ARB_transform_feedback2 (140) GL_ARB_transform_feedback3 (141) GL_ARB_transform_feedback_instanced (142) GL_ARB_transform_feedback_overflow_query (143) GL_ARB_transpose_matrix (144) GL_ARB_uniform_buffer_object (145) GL_ARB_vertex_array_bgra (146) GL_ARB_vertex_array_object (147) GL_ARB_vertex_attrib_64bit (148) GL_ARB_vertex_attrib_binding (149) GL_ARB_vertex_buffer_object (150) GL_ARB_vertex_program (151) GL_ARB_vertex_shader (152) GL_ARB_vertex_type_10f_11f_11f_rev (153) GL_ARB_vertex_type_2_10_10_10_rev (154) GL_ARB_viewport_array (155) GL_ARB_window_pos (156) GL_ATI_separate_stencil (157) GL_EXT_abgr (158) GL_EXT_bgra (159) GL_EXT_blend_color (160) GL_EXT_blend_equation_separate (161) GL_EXT_blend_func_separate (162) GL_EXT_blend_minmax (163) GL_EXT_blend_subtract (164) GL_EXT_clip_volume_hint (165) GL_EXT_compiled_vertex_array (166) GL_EXT_direct_state_access (167) GL_EXT_draw_buffers2 (168) GL_EXT_draw_range_elements (169) GL_EXT_fog_coord (170) GL_EXT_framebuffer_blit (171) GL_EXT_framebuffer_multisample (172) GL_EXT_framebuffer_object (173) GL_EXT_geometry_shader4 (174) GL_EXT_gpu_program_parameters (175) GL_EXT_gpu_shader4 (176) GL_EXT_memory_object (177) GL_EXT_memory_object_win32 (178) GL_EXT_multi_draw_arrays (179) GL_EXT_packed_depth_stencil (180) GL_EXT_packed_float (181) GL_EXT_packed_pixels (182) GL_EXT_polygon_offset_clamp (183) GL_EXT_rescale_normal (184) GL_EXT_secondary_color (185) GL_EXT_semaphore (186) GL_EXT_semaphore_win32 (187) GL_EXT_separate_specular_color (188) GL_EXT_shader_framebuffer_fetch (189) GL_EXT_shader_integer_mix (190) GL_EXT_shadow_funcs (191) GL_EXT_stencil_two_side (192) GL_EXT_stencil_wrap (193) GL_EXT_texture3D (194) GL_EXT_texture_array (195) GL_EXT_texture_compression_s3tc (196) GL_EXT_texture_edge_clamp (197) GL_EXT_texture_env_add (198) GL_EXT_texture_env_combine (199) GL_EXT_texture_filter_anisotropic (200) GL_EXT_texture_integer (201) GL_EXT_texture_lod_bias (202) GL_EXT_texture_rectangle (203) GL_EXT_texture_sRGB (204) GL_EXT_texture_sRGB_decode (205) GL_EXT_texture_shared_exponent (206) GL_EXT_texture_snorm (207) GL_EXT_texture_storage (208) GL_EXT_texture_swizzle (209) GL_EXT_timer_query (210) GL_EXT_transform_feedback (211) GL_IBM_texture_mirrored_repeat (212) GL_INTEL_conservative_rasterization (213) GL_INTEL_fragment_shader_ordering (214) GL_INTEL_framebuffer_CMAA (215) GL_INTEL_map_texture (216) GL_INTEL_multi_rate_fragment_shader (217) GL_INTEL_performance_query (218) GL_KHR_blend_equation_advanced (219) GL_KHR_blend_equation_advanced_coherent (220) GL_KHR_context_flush_control (221) GL_KHR_debug (222) GL_KHR_no_error (223) GL_KHR_shader_subgroup (224) GL_KHR_shader_subgroup_arithmetic (225) GL_KHR_shader_subgroup_ballot (226) GL_KHR_shader_subgroup_basic (227) GL_KHR_shader_subgroup_clustered (228) GL_KHR_shader_subgroup_quad (229) GL_KHR_shader_subgroup_shuffle (230) GL_KHR_shader_subgroup_shuffle_relative (231) GL_KHR_shader_subgroup_vote (232) GL_KHR_texture_compression_astc_hdr (233) GL_KHR_texture_compression_astc_ldr (234) GL_NV_blend_square (235) GL_NV_conditional_render (236) GL_NV_primitive_restart (237) GL_NV_texgen_reflection (238) GL_SGIS_generate_mipmap (239) GL_SGIS_texture_edge_clamp (240) GL_SGIS_texture_lod (241) GL_SUN_multi_draw_arrays (242) GL_WIN_swap_hint (243) WGL_EXT_swap_control 

----------------------------------------
Bugs:

  Buggy VBO: False
  Buggy FBO rendering to multi-sampling texture: False
  Buggy FBO rendering to cube map texture: False
  Buggy swap buffers with non-standard glViewport: False
  Buggy 32-bit depth buffer: False
  Buggy GLSL gl_FrontFacing: False
  Buggy GLSL read varying: False
  Buggy Texture Size Above 2048: False
  Buggy Shader Bump Mapping num_steps: False

----------------------------------------
Current buffers bit depths:

  Color buffer (red, green, blue, alpha): 8 / 8 / 8 / 8
  Depth: 24
  Stencil: 8
  Double buffer: True
  Multi-sampling (full-screen anti-aliasing) possible: True
    Current: 8 samples per pixel

----------------------------------------
Limits:

  Max clip planes: 8
  Max lights (fixed-function limit): 8
  Max texture size: 16384
  Max viewport dims (width, height): 16384 16384
  Max texture units (shader pipeline): 32
  Max texture units (fixed-function): 2
  Max cube map texture size: 16384
  Max 3d texture size: 2048
  Max texture max anisotropy: 16
  Occlusion query counter bits: 64
  Max renderbuffer size: 16384
  Max line width: 1.00 31.00

----------------------------------------
Memory (in Kb):

  Total: 0 (unknown)
  Current: 0 (unknown)
  Current for VBO: 0 (unknown)
  Current for Textures: 0 (unknown)
  Current for Renderbuffers: 0 (unknown)

**Comment by michaliskambi** [comment]
Thank you very much for the video -- the issue was immediately clear thanks to this. You have the Toolbar at the top hidden, I guess you even have it hidden by "File -> Preferences -> Show Status and Toolbar" so it's easy to forget it even was there :)

And we changed the code of "Xxx Warnings" some time ago around this -- the button was now part of toolbar, aligned differently, so the button was never visible if the toolbar was not visible.

Fixed, I made the "Xxx Warnings" button visible regardless of whether the toolbar is visible or not. When the toolbar is not visible, the "Xxx Warnings" button moves to right-top corner, similar to older version.

Snapshots on https://jenkins.castle-engine.io/public/builds/view3dscene/ are already updated with this fix.

**Comment by elmkni** [comment]
Hi Michalis,

thank you for the quick fix!

I've tested the latest Jenkins compile of [View3DSceme](https://jenkins.castle-engine.io/public/builds/view3dscene/view3dscene-4.3.0-win64-x86_64.zip) from 27-Apr-2023 22:21 (actually it is from 28-Apr-2023 00:20):

## Indeed issue #53 is gone and the warning button is back!
![view3dscene_issue_53](https://user-images.githubusercontent.com/53652478/235050621-b1afd7ec-e331-47d6-9fed-82f90da692b6.png)

Now I can wait in _patience_ for [issue #52](https://github.com/castle-engine/view3dscene/issues/52) (PixelTexture3D as emissive texture) to be fixed before the release of CGE 7.0 final ;)

With best regards,

Elmar
